pkg/gardenadm/botanist/botanist.go

```diff
@@ -223,12 +231,34 @@ func (b *AutonomousBotanist) initializeFakeGardenResources(ctx context.Context,
 		}
 	}
 
+	return initializeFakeGardenSecrets(ctx, b.GardenClient, secrets, secretBinding, credentialsBinding)
+}
+
+func initializeFakeGardenSecrets(
+	ctx context.Context,
+	gardenClient client.Client,
+	secrets []*corev1.Secret,
+	secretBinding *gardencorev1beta1.SecretBinding,
+	credentialsBinding *gardensecurityv1alpha1.CredentialsBinding,
```

Comment:
You could simplify this by passing a `[]client.Object` param and creating all objects in the slice. All object types seem to be treated equally.